test/e2e/framework/crd_util.go

```diff
@@ -0,0 +1,126 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package framework
+
+import (
+	"fmt"
+
+	apiextensionsv1beta1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1beta1"
+	crdclientset "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apiextensions-apiserver/test/integration/testserver"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/dynamic"
+)
+
+// CleanCrdFn declares the clean up function needed to remove the CRD
+type CleanCrdFn func() error
+
+// TestCrd holds all the pieces needed to test with the CRD
+type TestCrd struct {
+	Name               string
+	Kind               string
+	ApiGroup           string
+	ApiVersion         string
+	ApiExtensionClient *crdclientset.Clientset
+	Crd                *apiextensionsv1beta1.CustomResourceDefinition
+	DynamicClient      dynamic.ResourceInterface
+	CleanUp            CleanCrdFn
+}
+
+// CreateTestCRD creates a new CRD specifically for the calling test.
+func CreateTestCRD(f *Framework) (*TestCrd, error) {
+	suffix := randomSuffix()
```

Comment:
Is it okay that this will have some (very unlikely) collisions? The function just does this ```strconv.Itoa(r.Int() % 10000)```
I'm guessing yes, because it is still definitely an improvement to the flake rate. I guess I'm just wondering why framework doesn't have a way to generate suffixes that won't ever collide.

Reply:
Actually I am not sure it's okay to have very unlikely collisions here. However I would argue that it is the right thing to use randomSuffix() is a utility from the framework meant to solve this problem. The correct thing to do is to use it and we should probably file a bug to fix this. I do not believe this PR is the right place to fix that bug.